### PR TITLE
feat(web): A2-594 layout of interested party comments page

### PR DESCRIPTION
**Pull request by glpins**
## Describe your changes

IP Comments page

- tabs with awaiting review, valid and invalid comments
- pagination built in, but not applied. Page size hardcoded at 1000

Related

- comments test data added for S78 appeals
- IP comments row for appeals details built, but not applied. Will be handled in a separate ticket. Needs S78 feature flagging

[A2-594](https://pins-ds.atlassian.net/browse/A2-594)

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

[A2-456]: https://pins-ds.atlassian.net/browse/A2-456?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[A2-585]: https://pins-ds.atlassian.net/browse/A2-585?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[A2-594]: https://pins-ds.atlassian.net/browse/A2-594?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: appeals/api/src/server/repositories/representation.repository.js [hunk 1]
```diff
@@ -61,25 +61,33 @@ export const getFinalCommentsByAppealId = (appealId) => {
...
-		},
+	// Define the where clause for filtering
+	const whereClause = {
+		appealId,
+		representationType: APPEAL_REPRESENTATION_TYPE.COMMENT,
+		...(status && { status })
+	};
+
+	// Fetch the total count of matching representations
+	const itemCount = await databaseConnector.representation.count({
```
Comment (fabfab-kc) [inline appeals/api/src/server/repositories/representation.repository.js:84]: This is good, but a better way is to make a single roundtrip to the DB, so you can wrap in a transaction, e.g.:

```
const transaction = [
  databaseConnector.representation.count({...}),
  databaseConnector.representation.findMany({...})
];

return databaseConnector.$transaction(transaction)
```

**Review by fabfab-kc — CHANGES_REQUESTED** [review]
Looks good overall, just a suggestion to improve performance in the repository...

File: appeals/api/src/server/repositories/representation.repository.js (hunk 1 above)
Comment (BiggsKC) [reply appeals/api/src/server/repositories/representation.repository.js]: sorted

**Review by fabfab-kc — APPROVED** [review]
LGTM!

**Pull request merged**